common/src/main/java/org/figuramc/figura/mixin/render/renderers/SkullBlockRendererMixin.java

```diff
@@ -51,9 +51,8 @@ private static void renderSkull(Direction direction, float yaw, float animationP
         SkullBlockRendererAccessor.SkullRenderMode localMode = SkullBlockRendererAccessor.getRenderMode();
         SkullBlockRendererAccessor.setRenderMode(SkullBlockRendererAccessor.SkullRenderMode.OTHER);
 
-        // avatar
+        // avatar pointer incase avatar variable is set during render. (unlikely)
         Avatar localAvatar = avatar;
-        avatar = null;
```

Comment:
During testing avatar was  null, the permission field was null because the avatar field was null.
"java.lang.NullPointerException: Cannot read field "permissions" because "net.minecraft.class_836.avatar" is null"
We still encountered crashes during testing with just the permissions check.

Reply:
I see, what about making a copy in shouldRenderOffscreen instead? Did you try that

Reply:
No, i did not try that. It seems odd to use that as a fix, but i guess it would technically work.

Reply:
It seems to work. pushed changes here.